misc.go

```diff
@@ -37,6 +37,45 @@ func (e *RateLimitedError) Error() string {
 	return fmt.Sprintf("Slack rate limit exceeded, retry after %s", e.RetryAfter)
 }
 
+func getFile(ctx context.Context, client HTTPRequester, url, token string, writer io.Writer) error {
+	body := &bytes.Buffer{}
+
+	req, err := http.NewRequest("GET", url, body)
```

Comment:
this will likely segfault if NewRequest actually errors out, as req will be nil.